### ISSUE TITLE
ControlPersist breaks pssh and pscp

### ISSUE THREAD

**Issue by GoogleCodeExporter** [lilydjwg/pssh#67]
```
pssh and pscp work pretty well until I start using ControlPersist in 
~/.ssh/config:

Host *
   ControlMaster auto
   ControlPersist 30
   ControlPath ~/.ssh/master-%r@%h:%p

and then they always take more than 30 seconds to complete.

$ time pssh -H "$hosts" date
[1] 21:33:34 [SUCCESS] s1n1
[2] 21:33:34 [SUCCESS] s1n2
[3] 21:33:34 [SUCCESS] s2n1
[4] 21:33:34 [SUCCESS] s2n2
pssh -H "$hosts" date  0.07s user 0.05s system 0% cpu 30.671 total

$ time (for host in $hosts; do ssh $host 'echo `hostname`: `date`'; done)
s1n1: Fri Mar 2 21:34:10 GMT 2012
s1n2: Fri Mar 2 21:34:13 GMT 2012
s2n1: Fri Mar 2 21:34:11 UTC 2012
s2n2: Fri Mar 2 21:34:11 UTC 2012
( for host in $hosts; do; ssh $host 'echo `hostname`: `date`'; done; )  0.05s 
user 0.05s system 8% cpu 1.120 total

After a bit of experimentation, it is pretty clear that pssh does not complete 
until the newly created backgrounded ssh connections terminate.  A valid 
workaround is:

  export PSSH_OPTIONS='ControlPersist=no'

but I don't see why it shouldn't be able to work properly with ControlPersist 
enabled.
```

Original issue reported on code.google.com by `adam.spi...@gmail.com` on 2 Mar 2012 at 9:45

**Comment by richarson** [comment]
FWIW, this seems to be solved, at least with openssh-8.8p1 and pssh-2.3.4 on my system.

Runs with 8 hosts and `time pssh -p 5 -t 60 -h hosts-list -i 'uptime'` without Control* options give me consistent 4/5 seconds.

A run with 8 hosts and `time pssh -p 5 -t 60 -h hosts-list -i 'uptime'` with Control* options set also gives me 4/5 seconds, but subsequent runs give me consistent 0.2/0.3 seconds.

**Comment by lilydjwg** [comment]
Thanks, this actually has been working for quite some time here, but I didn't examine old exported issues. Closing.
